### ISSUE TITLE
[BUG] “An NVIDIA GPU is required to run tasks for this project” with GPUGRID

### ISSUE THREAD

**Issue by alexisLefebvre** [linuxserver/docker-boinc#25]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

> An NVIDIA GPU is required to run tasks for this project

I don't understand this part of the documentation: https://github.com/linuxserver/docker-boinc/blob/master/README.md#nvidia-gpu-support

- what is Zink?
- what is `nvidia-ctk`? Must it be installed on the host system?
- I tried the `deploy:` conf but it's not compatible with my version of `docker-compose`

The container can see the GPU:

```
docker-compose exec boinc bash -c "ls -lh /dev/dri/"
total 0
crw-rw---- 1 root video     226,   0 Nov 22 21:31 card0
crw-rw---- 1 root videooq54 226, 128 Nov 22 21:31 renderD128
```

### Expected Behavior

BOINC should be able to run tasks on the GPU.

### Steps To Reproduce

Open BOINC, add the GPUGRID project and try to get tasks.

### Environment

```markdown
- OS: Debian 12 (bookworm)
- How docker service was installed: I use `docker-compose` provided by Debian:


docker-compose --version
docker-compose version 1.25.0, build 0a186604
```
```


### CPU architecture

x86-64

### Docker creation

```bash
And this `docker-compose.yml` file:


---
version: "2.1"
services:
  boinc:
    image: lscr.io/linuxserver/boinc:latest
    container_name: boinc
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=Europe/Paris
#      - DRINODE=/dev/dri/renderD128
    volumes:
      - /var/docker/boinc:/config
    devices:
      - /dev/dri:/dev/dri
    ports:
      - 8090:8080
      - 8191:8181
    restart: unless-stopped
```
```


### Container logs

```bash
boinc    | 22-Nov-2024 21:36:37 [GPUGRID] Sending scheduler request: To fetch work.
boinc    | 22-Nov-2024 21:36:37 [GPUGRID] Requesting new tasks for CPU
boinc    | 22-Nov-2024 21:36:43 [GPUGRID] Scheduler request completed: got 0 new tasks
boinc    | 22-Nov-2024 21:36:43 [GPUGRID] An NVIDIA GPU is required to run tasks for this project
boinc    | 22-Nov-2024 21:36:43 [GPUGRID] No tasks sent
boinc    | 22-Nov-2024 21:36:43 [GPUGRID] Project requested delay of 11 seconds
boinc    | 22-Nov-2024 21:36:43 [GPUGRID] Warning: no_rsc_apps flag set for all resources.  Clearing flags.
boinc    | 22-Nov-2024 21:46:38 [GPUGRID] Sending scheduler request: To fetch work.
boinc    | 22-Nov-2024 21:46:38 [GPUGRID] Requesting new tasks for CPU
boinc    | 22-Nov-2024 21:46:44 [GPUGRID] Scheduler request completed: got 0 new tasks
boinc    | 22-Nov-2024 21:46:44 [GPUGRID] An NVIDIA GPU is required to run tasks for this project
boinc    | 22-Nov-2024 21:46:44 [GPUGRID] No tasks sent
boinc    | 22-Nov-2024 21:46:44 [GPUGRID] Project requested delay of 11 seconds
boinc    | 22-Nov-2024 21:46:44 [GPUGRID] Warning: no_rsc_apps flag set for all resources.  Clearing flags.
```

**Comment by aptalca** [comment]
Our readme does not list a version for the compose: https://github.com/linuxserver/docker-boinc/blob/master/README.md#docker-compose-recommended-click-here-for-more-info
You shouldn't be setting one.

Our readme got messed up during the templating.

See here for the original instructions for making the gpu available in this container: https://github.com/linuxserver/docker-boinc/commit/4851e8b50c5c3bf38847eb5a8966345c9211d225#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L78

**Comment by alexisLefebvre** [comment]
Thanks for your answer.

I switched to BOINC provided by Debian on the host system, without Docker, and GPU was recognized after using `apt install nvidia-driver`.

It looks like using a GPU from a Docker container requires several extra steps:

- https://github.com/NVIDIA/nvidia-docker has been replaced with https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html

I'll stick to the native BOINC client on the host.
